packages/cli/src/lib/implementation/config-middleware.ts

```diff
@@ -27,9 +32,22 @@ export async function configMiddleware<
         ...importedRc.upload,
         ...cliOptions.upload,
       },
+      // we can't use a async rc file as yargs does not support it. see: https://github.com/yargs/yargs/issues/2234
```

Comment:
In essence this is what made the whole PR a bis messy... I have an idea on how to fix it and will bring it up next meeting.

Reply:
OK, but honestly I like this implementation :grin: It's very explicit, therefore easy to understand how it works :+1: